### PR TITLE
Clustering map plugin

### PR DESCRIPTION
**Pull request by czhangsigmacomputing**
Customer requested plugin, which use deck.gl's IconClustering technique.

Full context is at https://sigmacomputing.atlassian.net/browse/SIG-19426

### CODE REVIEW

**Comment by czhangsigmacomputing** [comment]
@xyin96 and I also discussed the possibility to support filter for the plugin. It seems not trivial so I'll just describe the direction and leave it as a next stage work.

The idea is to use `window.parent.postMessage` to pass the filter info to workbook and set a global control to do the filter.
There are some functionality need to be supported before to make it work:
1. The event will be handled in `handleEvent.ts`, which require the exist of `elementId`, so either the `elementId` should be known by the plugin or there needs to be a new plugin function to help register an event handler. Also, the `wbOrigin` may also need if we don't want to pass in `*` to `postMessage` as the origin.
2. The most close thing to update global control like this is [here](https://github.com/sigmacomputing/slate/blob/f728679ed4c443043e7e7504d8e791c9a847b7dd/packages/slate/src/views/workbook/utils/index.ts#L214) (e.g where we use `workbook:variables:update`). But that part is specially used by embed case and seems not connected with plugin code. Therefore, need to add some new logic to make plugin logic work.

File: clustering-map/src/App.js [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+  useElementColumns,
+  useElementData,
+} from "@sigmacomputing/plugin";
+
+import './App.css';
+import Map from './components/Map';
+
+client.config.configureEditorPanel([
+  { name: "source", type: "element" },
+  { name: "Latitude", type: "column", source: "source", allowMultiple: false },
```
Comment (xyin96) [inline clustering-map/src/App.js:14]: You could limit the column types this config accepts with `allowedTypes`

Comment (czhangsigmacomputing) [reply clustering-map/src/App.js]: Thanks for the suggestion, added!

File: clustering-map/src/App.js [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+import './App.css';
+import Map from './components/Map';
+
+client.config.configureEditorPanel([
+  { name: "source", type: "element" },
+  { name: "Latitude", type: "column", source: "source", allowMultiple: false, allowedTypes: ['number', 'integer'] },
+  { name: "Longitude", type: "column", source: "source", allowMultiple: false, allowedTypes: ['number', 'integer'] },
+  { name: "Fields", type: "column", source: "source", allowMultiple: true },
+  { name: "Cluster scale", type: "text", placeholder: '1 to 100' },
```
Comment (xyin96) [inline clustering-map/src/App.js:17]: you can name the config `clusterScale` or something code-friendlier, and use the `label` property for display.
```suggestion
  { name: "cluster-scale", label: "Cluster scale", type: "text", placeholder: '1 to 100' },
```

**Review by xyin96 — APPROVED** [review]

**Pull request merged**